SmartcarAuth/SmartcarAuth.swift

```diff
@@ -144,19 +143,34 @@ Smartcar Authentication SDK for iOS written in Swift 3.
         let urlComp = URLComponents(url: url, resolvingAgainstBaseURL: false)
 
         guard let query = urlComp?.queryItems else {
-            return completion(AuthorizationError.missingQueryParameters, nil, nil)
+            return completion(AuthorizationError.missingQueryParameters, nil, nil, nil)
         }
 
         let queryState = query.filter({ $0.name == "state"}).first?.value
-
-        if query.filter({ $0.name == "error"}).first?.value != nil {
-            return completion(AuthorizationError.accessDenied, nil, queryState)
+        let vehicle = VehicleInfo()
+        vehicle.vin = query.filter({ $0.name == "vin"}).first?.value ?? ""
+        vehicle.make = query.filter({ $0.name == "make"}).first?.value ?? ""
+        vehicle.model = query.filter({ $0.name == "model"}).first?.value ?? ""
+        if let year = query.filter({ $0.name == "year"}).first?.value as? Int? {
+            vehicle.year = year
+        }
```

Comment:
This doesn't quite meet the intended spec for how vehicleInfo should be returned. With this logic, vehicleInfo is always returned, sometimes with junk values. 

What should happen is that vehicleInfo is only returned if vehicle information exists. 

Based on how we specced this feature, VehicleInfo is optional, but when it is returned, vin, make, and year are required fields (with model being optional).

[Check out how I implemented this in the JavaScript SDK](https://github.com/smartcar/javascript-sdk/blob/6e1b60a3f956f75f55ed3a43a637784b62df4235/src/sdk.js#L67). Notice how I return null for vehicleInfo when it doesn't exist?

Reply:
refactored it with you and kyle